### ISSUE TITLE
Packet SSH key resources consistently raising errors on creation attempt

### ISSUE THREAD

**Issue by Mierdin** [terraform-providers/terraform-provider-packet#154]
### Terraform Version

`Terraform v0.12.1`

### Affected Resource(s)

- packet_ssh_key
- packet_project_ ssh_key

### Terraform Configuration Files

Using the [example in the docs](https://www.terraform.io/docs/providers/packet/r/ssh_key.html)
as guidance:

```hcl
resource "packet_device" "lily" {
  hostname         = "lily"
  plan             = "t1.small.x86"
  facilities = ["sjc1"]
  operating_system = "centos_7"
  billing_cycle    = "hourly"
  project_id       = "${var.packet_project_id}"
  depends_on       = ["packet_ssh_key.mierdinkey"]
}

resource "packet_ssh_key" "mierdinkey" {
  name       = "mierdinkey"
  public_key = "<omitted>"
}
```

### Debug Output

[This is the output](https://gist.github.com/Mierdin/f590c74c1498974d2783dd8602d23979) I see from two subsequent runs of `terraform apply`. In the first run, the provider errors out, because the key doesn't exist, which is true, but it succeeds in creating it before the error, apparently. This apparently causes the second error in the subsequent run of `terraform apply`, which indicates the key already exists.

### Panic Output

N/A

### Expected Behavior

Key resources should be created if needed, or if they already exist because of previous `terraform apply` commands, should ignore silently.

### Actual Behavior

On the first run of `terraform apply`, I get an error that the key doesn't exist. This is obviously true, since I am expecting the provider to create it in the first place. Interestingly enough, it *does*, but then errors, which prevents me from being able to create compute resources.

In addition, a second subsequent run of `terraform apply` with no changes, produces an additional error that the key already exists - which is true because the last run succeeded in creating the key even if it produced an error. Again, this error prohibits me from continuing.

Similar behavior happens for both project-level keys, or user keys.

### Steps to Reproduce

1. Run `terraform apply` with the above examples.

### Important Factoids

New Packet account, so no chance of conflicting resources, etc - totally greenfield.

### References

> Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? 

None that I could find, but I am happy to be corrected.

**Comment by t0mk** [comment]
hi @Mierdin, thanks for reporting, I wil take a look today.

**Comment by t0mk** [comment]
@Mierdin
- Your first `apply` is strange, the provider should not err when the key doesn't exist. It should just create it.
- The second apply is also strange, because if the key exists, the provider should just omit any action. This error (`Error: Key already exists`) might happen when you have already SSH key resource with the same `public_key` string  in the API, but under a different name.  It's an API sanity check.

I would really like to know what happens in the API requests. Do you think you could reproduce again, but this time with TF_LOG=DEBUG set. You can run the applies like this:
```
TF_LOG=DEBUG tf apply &>> log
```
.. it will print the HTTP requests and responses. Then you can upload the `log` file. Before you upload it, please remove lines containing `X-Auth-Token:` header, that's where you API token is, and we don't want to see that on github. Thanks!

**Comment by Mierdin** [comment]
Forgive the delayed response, I was working on this with Packet support. Nothing wrong with the provider, so I'll close this issue after a brief explanation. In short, I was using an API token defined at the project scope, and defining a user-level SSH key. Moving to a personal API token fixed this, but I still have some lingering thoughts:

- The error message provided is very non-descriptive regarding the actual cause of the problem. If I'm using a project-level key and trying to change a personal SSH key, it makes sense this would error because it wouldn't know which user I was creating a key for - however, the error message is very generic and doesn't lead naturally to this cause.
- There are a number of API endpoints that work by indirect parameters, such as the currently logged-in user, which are equally accessible by project API tokens, but that might result in the aforementioned error condition. I much prefer the kind of explicit parameterization present in the [project-level](https://www.packet.com/developers/api/#retrieve-a-projects-ssk-keys) endpoints.

So, TL;DR: use personal API tokens when using this provider. In any case, this is all fairly minor and I'm mostly nitpicking. Thanks for working on this provider, and keep up the good work!

**Comment by ognots** [comment]
I've experienced this and thought the provider was straight up broken. the error message should definitely mention that personal api keys must be used.
As a secondary note, this makes running this provider from CI less than ideal. generally we use project api keys rather than personal ones in CI
